cloud/services/routetables/routetables_test.go

```diff
@@ -115,7 +116,8 @@ func TestReconcileRouteTables(t *testing.T) {
 			},
 			expectedError: "",
 			expect: func(m *mock_routetables.MockClientMockRecorder) {
-				m.CreateOrUpdate(context.TODO(), "my-rg", "my-routetable", gomock.AssignableToTypeOf(network.RouteTable{}))
```

Comment:
the test was missing the `defer mockCtrl.Finish()` (added below) so this was not being checked. The expectation for this test should be that the route table createorupdate is *not* called.  Also fixed this for the rest of the tests in this file and added explicit checks for number of times functions are expected to be called.

Reply:
created #720 to look into other tests where this might be the case

Reply:
Great catch!